Network/HTTP/Req.hs

```diff
@@ -937,8 +960,49 @@ uriHost uri = case URI.uriAuthority uri of
   Right URI.Authority {..} ->
     Just (URI.unRText authHost)
 
+-- | A quasiquoter to build a 'Url'. Can be used both as an expression and as a
+-- pattern. Only works for URIs that don't involve query options (e.g. port)
+--
+-- >>> [urlQ|https://exmaple.org/test|]
+-- Url Https ("test" :| ["example.org"])
+urlQ :: TH.QuasiQuoter
+urlQ = TH.QuasiQuoter
```

Comment:
Why do we need two different quoters for URLs with and without options?

Reply:
Because I assumed that the most common case would be that `Option` is `mempty` in which case we just want the url and not a url and option tuple. With just one quasiquoter the options would be:

- Always return a url and options tuple. I don't like this, because most of the time you will just directly extract the `Url` and ignore the `Option` which you know (or think you know) is `mempty`. However maybe you're wrong. I wasn't aware the port number was an option. `urlQ` like I implemented it will let you know, because it causes a type error if the `Option` value isn't mempty.
- Return a url if we don't have any options and return a tuple if we do. Sounds rather confusing to me, but if you prefer that behavior I could change that.

Reply:
I think the way I'd use it is with `let` or `where` binding to pattern match on the tuple, actually. Having two quoters unnecessarily increases the complexity IMO.

Reply:
Yes, again, but then you can fall into the trap of assuming the `Option` part is `mempty` and just ignoring it, when it actually isn't.

Reply:
I would never assume that. Discarding part of output like that is asking for troubles, unless you really know why you do it.

Reply:
Yes exactly, which is why this gives you a type safe way to “discard” that part if it 'mempty'.

Reply:
```
>>> [urlQ|https://example.org:443/|]

<interactive>:3:7: error:
    • URL with options not supported, please use urlWithOptsQ
    • In the quasi-quotation: [urlQ|https://example.org:443/|]

```

Reply:
So you want a type-safe guarantee that what you've just written yourself as a literal URI in the quasiquoter doesn't have anything that would contribute to the `Option` component? It is so much simpler, in my opinion, to just assume that it may always have `Option` component and force the user to plug the `Option` part where it belongs.

What I do not like is having different ways of doing the same thing while all we gain is very questionable ergonomic value.

Reply:
If you are really adamant about having two of them, I'll take my time and think about it more.

Reply:
I'm not, maybe you're right. You save yourself a let. Is that worth a second quoter? Maybe not.
```
req GET [urlQ|https://example.org/|] NoReqBody bsResponse mempty

let (url, opts) = [urlQ|https://example.org/|]
in req GET url NoReqBody bsResponse opts
```